doc/realtime_servo/realtime_servo_tutorial.rst

```diff
@@ -1,134 +1,123 @@
+    <video width="700px" nocontrols="true" autoplay="true" loop="true">
+        <source src="../../_static/Servo_Teleop_Demo.webm" type="video/webm">
+        Teleoperation demo with controller
+    </video>
 
-    rostopic pub -r 100 -s /servo_server/delta_twist_cmds geometry_msgs/TwistStamped "header: auto
-    twist:
-      linear:
-        x: 0.0
-        y: 0.01
-        z: -0.01
-      angular:
-        x: 0.0
-        y: 0.0
-        z: 0.0"
+Note that the controller overlay here is just for demonstration purposes and is not actually included
 
-The `-r 100` sends new commands at a 100 Hz rate. The combination of `-s` and `auto` updates the timestamp automatically.
+Using the C++ Interface
+-----------------------
+Instead of launching Servo as its own component, you can include Servo in your own nodes via the C++ interface. Sending commands to the robot is very similiar in both cases, but for the C++ interface a little bit of setup for Servo is necessary. In exchange, you will be able to directly interact with Servo through its C++ API.
 
-Settings
---------
-User-configurable settings of the servo node are well-documented in ``servo/config/ur_simulated_config.yaml``.
+This basic C++ interface demo moves the robot in a predetermined way and can be launched with ::
 
-Changing Control Dimensions
----------------------------
-The dimensions being controlled may be changed with the ``change_control_dimensions`` (``moveit_msgs/ChangeControlDimensions``) service. The service consists of the 6 booleans representing the manipulator's degrees of freedom (3 translation and 3 rotation), defined in the input velocity command frame. Setting one of these booleans to ``false`` will result in the user input in that direction being overwritten with a 0, thus disallowing motion in the direction.
+    ros2 launch moveit2_tutorials servo_cpp_interface_demo.launch.py
 
-Robot Requirements
-------------------
-The servo node streams an array of position or velocity commands to the robot controller. This is compatible with ros\_control ``position_controllers/JointGroupPositionControllers`` or ``velocity_controllers/JointGroupVelocityControllers``. You can check if these controllers are available for your robot by searching for the controller config file (typically named ``controllers.yaml``). After launching the robot, you can check if any ros_control controllers are available with: ::
+An Rviz window should appear with a Panda arm and collision object. The arm will joint-jog for a few seconds before switching to a Cartesian movement. As the arm approaches the collision object, it slows and stops.
 
-    rosservice call /controller_manager/list_controllers
+Expected Output
+---------------
+.. raw:: html
 
-And switch to the desired controller with: ::
+    <video width="700px" nocontrols="true" autoplay="true" loop="true">
+        <source src="../../_static/C++_Interface_Demo.webm" type="video/webm">
```

Comment:
This video wont play on Chrome for me, but works with Firefox.

Reply:
You saw this when you built locally right? I am unsure how to resolve this, but I did notice that another gif/video in the RViz Quickstart tutorial does not play on Chrome when I build locally, but it does play if accessed through [the webpage](http://moveit2_tutorials.picknik.ai/doc/quickstart_in_rviz/quickstart_in_rviz_tutorial.html#moving-joints-or-in-null-space)

Reply:
Yes I only checked on a local build.